config/daemonconfig/fuse.go

```diff
@@ -94,9 +95,36 @@ func (c *FuseDaemonConfig) DumpString() (string, error) {
 	return DumpConfigString(c)
 }
 
+// We can set the Backend tag to "-" so that Backend Config does not write to local files.
+// But it needs to be compatible with the old version,
+// and it is not possible to control whether Backend Config is written to the local file through tag.
+// Therefore, a deep copy of the Config information is made here,
+// and the Backend Config is set to null.
 func (c *FuseDaemonConfig) DumpFile(f string) error {
 	if err := os.MkdirAll(path.Dir(f), 0755); err != nil {
 		return err
 	}
+	if config.IsCredentialSourceEnabled() {
+		switch c.Device.Backend.BackendType {
+		case BackendTypeRegistry:
+			backendAuth := c.Device.Backend.Config.Auth
+			registryToken := c.Device.Backend.Config.RegistryToken
+			c.Device.Backend.Config.Auth = ""
+			c.Device.Backend.Config.RegistryToken = ""
+			err := DumpConfigFile(c, f)
+			c.Device.Backend.Config.Auth = backendAuth
+			c.Device.Backend.Config.RegistryToken = registryToken
+			return err
+		case BackendTypeOss:
```

Comment:
The `s3` backend should also be handled. :)

Reply:
Perhaps a more elegant way would be to use struct tags and reflect, for example:

```
type BackendConfig struct {
	...
	Auth               string         `json:"auth,omitempty" secret:"true"`
	RegistryToken      string         `json:"registry_token,omitempty" secret:"true"`

	AccessKeyID     string `json:"access_key_id,omitempty" secret:"true"`
	AccessKeySecret string `json:"access_key_secret,omitempty" secret:"true"`
  ...
}
```

Reply:
> The `s3` backend should also be handled. :)

Sorry for this, the `s3` backend only be added in https://github.com/dragonflyoss/image-service/blob/631db29759f4dceacaa7ff80b845a9d0107beef5/api/src/config.rs#L520, not in snapshshotter, maybe we don't need to handle it first.